filters/filters.go

```diff
@@ -38,10 +38,13 @@ type FilterContext interface {
 	// route.
 	Served() bool
```

Comment:
The proxy needs to check if the request has been served already. Should we switch to another strategy like, checking if there is already a Response object?